### PR TITLE
Remove usage of deprecated VS Code Didact

### PR DESCRIPTION
**Pull request by tdiesler**
Update generic languages, env, cron

### CODE REVIEW

**Review by squakez — APPROVED** [review]
Cool cleanup, thanks!

File: .vscode/extensions.json [hunk 1]
```diff
@@ -1,6 +0,0 @@
-{
-  "recommendations": [
-    "redhat.vscode-didact",
-    "redhat.apache-camel-extension-pack"
```
Comment (apupier) [inline .vscode/extensions.json:4]: What is the reason to remove the extension pack for Camel?

**Review by squakez — COMMENTED** [review]
I think you forgot to swith `resource-file-xyz` examples. You've also removing `resources-data` files which were used by those examples.

**Comment by tdiesler** [comment]
This [commit](https://github.com/apache/camel-k/commit/7ee129afcd7ee53b4f25f5dab1ecc55689b1f2cd) removes `resource-file-xyz.groovy` route definitions without providing yaml equivalents. This was because none of the tests referenced those groovy files. Perhaps the examples provide close enough testing templates to resurrect `resource-file` testing.

**Comment by squakez** [comment]
> This [commit](https://github.com/apache/camel-k/commit/7ee129afcd7ee53b4f25f5dab1ecc55689b1f2cd) removes `resource-file-xyz.groovy` route definitions without providing yaml equivalents. This was because none of the tests referenced those groovy files. Perhaps the examples provide close enough testing templates to resurrect `resource-file` testing.
> 
> We don't have `--resource file:*` (any more), only `--resource configmap:*` and `--resource secret:*`

We don't need to have parity between tests and examples. Examples are typically more oriented to show basic or complex use cases for users. I think in this case we do need to convert the binary and base64 from groovy to yaml and moving also from file to configmap (if no other similar example is already available, of course) as it is a useful demo scenario to show.

**Comment by tdiesler** [comment]
I now migrated and restored all `config/resource-file-xyz` examples

**Comment by tdiesler** [comment]
This is ready to merge

**Review by squakez — COMMENTED** [review]
I've just noticed something that would require some attention. We're missing to add some comment line to explain in the same example how to run it. For example, in the groovy example we used to have:
```

//
// To run this integrations use:
// 
// kubectl create configmap my-cm --from-literal=my-configmap-key="configmap content"
// kamel run --resource configmap:my-cm@/tmp/app/data resource-configmap-location-route.groovy --dev
//
```
And now we're missing it in the new example. I think we must bring this comment back in the new specific dsl format, of course.

**Pull request merged**